### PR TITLE
Vcsim container

### PR DESCRIPTION
**Pull request by jctanner**
##### SUMMARY

Adds a container that simulates the vcenter API for use in integration testing vmware modules.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
integration tests

##### ANSIBLE VERSION

```
2.4
```

### CODE REVIEW

**Comment by jctanner** [comment]
cc @dav1x @nerzhul @dag

**Comment by jctanner** [comment]
Here's an example test that will show up in a future PR that makes use of this container ...

```
- name: store the vcenter container ip                                                                                                                                                                                                 
  set_fact:
    vcsim: "{{ lookup('env', 'vcenter_host') }}"
- debug: var=vcsim

- name: kill vcsim
  uri:
    url: "{{ 'http://' + vcsim + ':5000/killall' }}"
- name: start vcsim
  uri:
    url: "{{ 'http://' + vcsim + ':5000/spawn' }}"
  register: vcsim_instance

#- name: make sure pyvmomi is installed
#  package:
#    name: python-pyvmomi
#    state: latest    

- name: get a list of VMS from vcsim   
  uri:
    url: "{{ 'http://' + vcsim + ':5000/govc_find?filter=_VM' }}"
  register: vmlist

- debug: var=vcsim_instance
- debug: var=vmlist

- name: show the  
  debug:
    msg: "{{ (item|basename).split('_')[0] }}"
  with_items: "{{ vmlist['json'] }}"

- name: find folders for each vm
  vmware_guest_find:
    validate_certs: False
    hostname: "{{ vcsim_instance['json']['host'] }}"
    username: "{{ vcsim_instance['json']['username'] }}"
    password: "{{ vcsim_instance['json']['password'] }}"
    name: "{{ item|basename }}"
    datacenter: "{{ (item|basename).split('_')[0] }}"
    #folder: /F0/DC0/F0/DC0/vm/F0
    #folder: /vm/F0
    #folder: /F0
  with_items: "{{ vmlist['json'] }}"


- name: get facts for each vm
  vmware_guest_facts:
    validate_certs: False
    hostname: "{{ vcsim_instance['json']['host'] }}"
    username: "{{ vcsim_instance['json']['username'] }}"
    password: "{{ vcsim_instance['json']['password'] }}"
    name: "{{ item|basename }}"
    datacenter: "{{ (item|basename).split('_')[0] }}"
    #folder: /F0/DC0/F0/DC0/vm/F0
    #folder: /vm/F0
    #folder: /F0
    folder: /F0/DC0/vm/F0
  with_items: "{{ vmlist['json'] }}"
```

File: test/utils/docker/vcenter-simulator/make.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+docker build -t jctanner:vcenter-simulator .
```
Comment (gundalow) [inline test/utils/docker/vcenter-simulator/make.sh]: Does this need moving over to the live version
`ansible:vcenter-simulator`

File: test/runner/lib/cloud/vcenter.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+"""CloudStack plugin for integration tests."""
```
Comment (gundalow) [inline test/runner/lib/cloud/vcenter.py]: s/CloudStack/VMware vCentre/

**Comment by jctanner** [comment]
vcsim is moving to a new repo, so I need to make sure the go install paths are eventually updated https://github.com/vmware/vic/pull/5531

**Pull request merged**